ibm/resource_ibm_cloudant.go

```diff
@@ -0,0 +1,586 @@
+}
+
+func resourceIBMCloudantUpdate(d *schema.ResourceData, meta interface{}) error {
+	d.Set("service", "cloudantnosqldb")
+
+	err := validateCloudantInstanceCapacity(d)
+	if err != nil {
+		return err
+	}
+
+	err = validateCloudantInstanceCors(d)
+	if err != nil {
+		return err
+	}
+
+	err = resourceIBMResourceInstanceUpdate(d, meta)
+	if err != nil {
+		return err
+	}
+
+	client, err := getCloudantClient(d, meta)
+	if err != nil {
+		return err
+	}
+
+	if d.HasChange("include_data_events") {
+		err := updateCloudantActivityTrackerEvents(client, d)
+		if err != nil {
+			return fmt.Errorf("Error updating activity tracker events: %s", err)
+		}
+	}
+
+	if d.HasChange("capacity") {
+		err := updateCloudantInstanceCapacity(client, d)
+		if err != nil {
+			return fmt.Errorf("Error retrieving capacity throughput information: %s", err)
+		}
+	}
+
+	if d.HasChange("enable_cors") {
+		err := updateCloudantInstanceCors(client, d)
+		if err != nil {
+			return fmt.Errorf("Error updating CORS settings: %s", err)
+		}
+	}
+
+	return resourceIBMCloudantRead(d, meta)
+}
+
+func setCloudantLegacyCredentials(d *schema.ResourceData, meta interface{}) error {
+	rsConClient, err := meta.(ClientSession).ResourceControllerV2API()
+	if err != nil {
+		return err
+	}
+
+	instanceID := d.Id()
+	resourceInstanceGet := rc.GetResourceInstanceOptions{
+		ID: &instanceID,
+	}
+
+	instance, response, err := rsConClient.GetResourceInstance(&resourceInstanceGet)
+	if err != nil {
+		log.Printf("[DEBUG] Error retrieving resource instance: %s\n%s", err, response)
+		return err
+	}
+
+	parameters := instance.Parameters
+
+	if legacyCredentials, ok := parameters["legacyCredentials"]; ok {
+		switch v := legacyCredentials.(type) {
+		case bool:
+			d.Set("legacy_credentials", v)
+		case string:
+			d.Set("legacy_credentials", v == "true")
+		default:
+			d.Set("legacy_credentials", false)
+		}
+	}
+
+	if environmentCRN, ok := parameters["environment_crn"]; ok {
+		v := environmentCRN.(string)
+		d.Set("environment_crn", v)
+	}
+
+	return nil
+}
+
+func setCloudantResourceControllerURL(d *schema.ResourceData, meta interface{}) error {
+	crn := d.Get(ResourceCRN).(string)
+	rcontroller, err := getBaseController(meta)
+	if err != nil {
+		return err
+	}
+	d.Set(ResourceControllerURL, rcontroller+"/services/cloudantnosqldb/"+url.QueryEscape(crn))
+
+	return nil
+}
+
+func getCloudantClient(d *schema.ResourceData, meta interface{}) (*cloudantv1.CloudantV1, error) {
+
+	session, err := meta.(ClientSession).BluemixSession()
+	if err != nil {
+		return nil, err
+	}
+
+	var endpoint string
+	extensions := d.Get("extensions").(map[string]interface{})
+	if v, ok := extensions["endpoints.public"]; ok {
+		endpoint = "https://" + v.(string)
+	}
+
+	switch session.Config.Visibility {
+	case "private":
+		_, ok := extensions["endpoints.private"]
+		if !ok {
+			return nil, fmt.Errorf("Missing endpoints.private in extensions")
+		}
+		endpoint = extensions["endpoints.private"].(string)
```

Comment:
When we set `endpoint` for the other cases we have to prepend `"https://"`, but we don't here. Is there actually a difference or is it an oversight?